hw/ip/spi_device/data/spi_device.hjson

```diff
@@ -270,17 +270,21 @@
         }
       ],
     },
+    { skipto: "0x1000" }
     { window: {
         name: "buffer",
-        items: "512",
+        items: "1024",
         validbits: "32",
         byte-write: "false",
         unusual: "false"
         swaccess: "rw",
         desc: '''
-          SPI internal 2kB buffer.
+          SPI internal buffer.
 
-          This buffer is shared by RX and TX circular buffer together.
+          In Generic mode, this buffer is used for RX/TX buffer.  In Flash &
+          Passthrough mode, lower 2kB is for Read content emulating eFlash.
+          next 1kB is for Mailbox read/write buffer. The rest is 64B of
+          CmdFIFO, 64B of AddrFIFO, and 256B of payload FIFO.
```

Comment:
The description here has changed significantly, does it mean that some more functionality has been added to this register that needs to be reflected in software?

If that is the case, then I think an issue should be opened and assigned to the original DIF owner, otherwise can be ignored.

Reply:
Right. Current generic mode shall remain as it is. Just it will introduce new modes (flash mode/ passthrough mode) and re-use the same DpSRAM content for other purposes. 

I think managing those multiple modes in the same DIF does not make sense as the new modes are almost exclusive from current SPI.